hub/metrics.go

```diff
@@ -8,6 +8,7 @@ import (
 )
 
 // Metrics store Hub collected metrics.
+// TODO(dunglas): check how to let Caddy collecting these metrics: https://caddyserver.com/docs/metrics
```

Comment:
IMHO metrics should be an interface. Then one can easily implement any kind of metrics collection (eg. implement an OpenTelemetry based metrics collection). That would increase the reusability of this library.

Reply:
Yes why not 👍, PR welcome on this too!

Reply:
Will do, once this PR is merged.